tests/PHPUnit/Integration/ArchiveProcessor/LoaderTest.php

```diff
@@ -191,16 +191,17 @@ public function testPluginOnlyArchivingDoesNotRelaunchChildArchivesWhenReusingAl
                 'date2' => '2020-01-20',
                 'period' => '1',
             ],
+            // archive 4 is missing as VisitsSummary is archived twice, as it doesn't contain data
```

Comment:
This is caused by some weird behavior of archiving. When archiving is triggered for anything, the first thing that is done is checking if core metrics (aka `VisitsSummary`) needs to be archived. This is done by checking if there is an archive that contains the visits metric. If this isn't the case `VisitsSummary` will be archived.
In case we have an archive that doesn't contain any visits, the archive will only contain a `done` flag, but no other metrics. This causes the archiving to be triggered again, which will create a new (empty) archive, while removing the previous one.
As archiving dependent segments first triggers archiving `VisitsSummary`, it creates an empty archive. Afterwards when archiving the Goals plugin it will archive `VisitsSummary` again, as the previous one is empty. Which then causes this missing archive id.

Reply:
Maybe add some of the explanation from the GH comment as part of the code comment?